src/heaps.jl

```diff
@@ -63,7 +63,14 @@ include("heaps/binary_heap.jl")
 include("heaps/mutable_binary_heap.jl")
 
 # generic functions
+"""
+    extract_heap!(h)
 
+### Args:
+* `h`: an abstract heap.
+
+This function returns an array which has the elements from the heap extracted.
```

Comment:
No need for "this function". Also, this string doesn't explain much: it merely repeats "extracted". IIUC, what's important here is that `r` is emptied.